src/dotnet/commands/dotnet-projectmodel-server/Messengers/GlobalErrorMessenger.cs

```diff
@@ -16,9 +16,12 @@ protected override bool CheckDifference(ProjectSnapshot local, ProjectSnapshot r
             return remote != null && Equals(local.GlobalErrorMessage, remote.GlobalErrorMessage);
         }
 
-        protected override object CreatePayload(ProjectSnapshot local)
+        protected override void SendPayload(ProjectSnapshot local, Action<object> send)
         {
-            return local.GlobalErrorMessage;
+            if (local.GlobalErrorMessage != null)
```

Comment:
There are other messengers use the pattern to create their own message. The logic of not sending `null` payload is specific to `GlobalErrorMessage`. For other messenger the logic may alter to send an empty message.